packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -678,10 +679,19 @@ class _JsonQuery implements Query<Map<String, dynamic>> {
         );
       }
 
+      if (field != FieldPath.documentId && hasDocumentIdField) {
+        assert(
+          operator != '!=',
+          'You cannot use ' !=
+              ' filters whilst using a FieldPath.documentId field in another filter.',
+        );
+      }
+
       if (field == FieldPath.documentId) {
         assert(
-          !hasNotEqualTo,
-          "You cannot use '!=' filters with a FieldPath.documentId field.",
+          !hasNotEqualToNotOperatorAndNotDocumentIdField,
+          'You cannot use FieldPath.documentId field whilst using a ' !=
```

Comment:
There's an issue with quotes I think. This will print "false" instead of the message